test/foundry-tests/SqrtPriceMath.t.sol

```diff
@@ -0,0 +1,384 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import "forge-std/console.sol";
+import {Test} from "forge-std/Test.sol";
+import {Vm} from "forge-std/Vm.sol";
+import {SqrtPriceMathTest} from "../../contracts/test/SqrtPriceMathTest.sol";
+import {Constants} from "./utils/Constants.sol";
+
+contract SqrtPriceMathTestTest is Test {
+    SqrtPriceMathTest sqrtPriceMath;
+
+    function setUp() public {
+        sqrtPriceMath = new SqrtPriceMathTest();
+    }
+
+    function expandTo18Decimals(uint256 a) internal pure returns (uint256) {
+        uint256 multiplier = 10 ** 18;
+        return a * multiplier;
+    }
+
+    function test_getNextSqrtPriceFromInput_revertsIfPriceIsZero() public {
+        vm.expectRevert();
+        sqrtPriceMath.getNextSqrtPriceFromInput(0, 0, expandTo18Decimals(1) / 10, false);
+    }
+
+    function test_getNextSqrtPriceFromInput_revertsIfLiquidityIsZero() public {
+        vm.expectRevert();
+        sqrtPriceMath.getNextSqrtPriceFromInput(1, 0, expandTo18Decimals(1) / 10, true);
+    }
+
+    function test_getNextSqrtPriceFromInput_revertsIfInputAmountOverflowsThePrice() public {
+        uint160 price = 2 ** 160 - 1;
+        uint128 liquidity = 1024;
+        uint256 amountIn = 1024;
+
+        vm.expectRevert();
+        sqrtPriceMath.getNextSqrtPriceFromInput(price, liquidity, amountIn, false);
+    }
+
+    function test_getNextSqrtPriceFromInput_anyInputAmountCannotUnderflowThePrice() public {
+        uint160 price = 1;
+        uint128 liquidity = 1;
+        uint256 amountIn = 2 ** 255;
+
+        assertEq(sqrtPriceMath.getNextSqrtPriceFromInput(price, liquidity, amountIn, true), 1);
+    }
+
+    function test_getNextSqrtPriceFromInput_returnsInputPriceIfAmountInIsZeroAndZeroForOneEqualsTrue() public {
+        uint160 price = Constants.SQRT_RATIO_1_1;
+        uint128 liquidity = 1;
+
+        assertEq(sqrtPriceMath.getNextSqrtPriceFromInput(price, liquidity, 0, true), price);
+    }
+
+    function test_getNextSqrtPriceFromInput_returnsInputPriceIfAmountInIsZeroAndZeroForOneEqualsFalse() public {
+        uint160 price = Constants.SQRT_RATIO_1_1;
+        uint128 liquidity = 1;
+
+        assertEq(sqrtPriceMath.getNextSqrtPriceFromInput(price, liquidity, 0, false), price);
+    }
+
+    function test_getNextSqrtPriceFromInput_returnsTheMinimumPriceForMaxInputs() public {
+        uint160 sqrtP = 2 ** 160 - 1;
+        uint128 liquidity = Constants.MAX_UINT128;
+        uint256 maxAmountNoOverflow = Constants.MAX_UINT256 - Constants.MAX_UINT128 << 96 / sqrtP;
+
+        assertEq(sqrtPriceMath.getNextSqrtPriceFromInput(sqrtP, liquidity, maxAmountNoOverflow, true), 1);
+    }
+
+    function test_getNextSqrtPriceFromInput_inputAmountOf0_1Currency1() public {
+        uint160 sqrtP = Constants.SQRT_RATIO_1_1; // sqrtP = encodeSqrtPriceX96(1, 1)
```

Comment:
don't need javascript reference comment here

Reply:
@ewilz thank you for the review! I will incorporate changes and push the updates soon.